--- a/src/video/fbcon/SDL_fbevents.c	2016-09-26 10:30:36.677628297 -0500
+++ b/src/video/fbcon/SDL_fbevents.c	2016-09-26 10:31:01.393628795 -0500
@@ -1020,6 +1020,28 @@
 				}
 				break;
 			}
+
+			/* Zipit Z2 Hack to handle switching VT's
+			 * This uses ALT-RIGHTARROW and ALT-LEFTARROW to cycle
+			 * up or down through VT's. The Zipit keymap is also
+			 * configured to switch VT's with the same combo so
+			 * this should work everywhere.
+			 */
+			case SDLK_LEFT:
+			if (SDL_GetModState() & KMOD_ALT) {
+				if ( pressed ) {
+					switch_vt(this, current_vt-1);
+				}
+				break;
+			}
+			case SDLK_RIGHT:
+			if (SDL_GetModState() & KMOD_ALT) {
+				if ( pressed ) {
+					switch_vt(this, current_vt+1);
+				}
+				break;
+			}
+
 			/* Fall through to normal processing */
 		    default:
 			posted += SDL_PrivateKeyboard(pressed, &keysym);
