--- navit-0.5.0/navit/graphics/sdl/graphics_sdl.c~	2015-12-31 02:42:41.000000000 -0500
+++ navit-0.5.0/navit/graphics/sdl/graphics_sdl.c	2016-10-09 09:48:18.569001397 -0400
@@ -1138,6 +1138,7 @@
 				break;
 			    }
 			case SDLK_PAGEUP:
+			case SDLK_HOME:
 			    {
 				keybuf[0] = NAVIT_KEY_ZOOM_OUT;
 				break;
@@ -1148,6 +1149,7 @@
 				break;
 			    }
 			case SDLK_PAGEDOWN:
+			case SDLK_END:
 			    {
 				keybuf[0] = NAVIT_KEY_ZOOM_IN;
 				break;
