--- gmu-0.10.1/src/frontends/sdl/plbrowser.c~	2015-05-15 12:05:50.000000000 -0400
+++ gmu-0.10.1/src/frontends/sdl/plbrowser.c	2016-10-23 11:33:44.196611487 -0400
@@ -119,8 +119,8 @@
 	}
 
 	pl_length = gmu_core_playlist_get_length();
-	snprintf(buf, 63, "Playlist (%d %s, mode: %s)", pl_length,
-	         pl_length != 1 ? "entries" : "entry", mode);
+	snprintf(buf, 63, "Playlist (%d %s, %s)", pl_length,
+	         pl_length != 1 ? "items" : "item", mode);
 	skin_draw_header_text(pb->skin, buf, sdl_target);
 
 	if (pb->first_visible_item == -1 || pb->offset == 0) {
