diff -rupN gmu-0.10.1.orig/src/tools/charset.h gmu-0.10.1/src/tools/charset.h
--- gmu-0.10.1.orig/src/tools/charset.h	1969-12-31 19:00:00.000000000 -0500
+++ gmu-0.10.1/src/tools/charset.h	2016-06-14 06:37:37.223404607 -0400
@@ -0,0 +1,41 @@
+#ifndef WEJ_CHARSET_H
+#define WEJ_CHARSET_H
+
+#ifdef USE_WIDE_CHAR
+#include <wctype.h>
+
+static char *wchars_to_utf8_str_realloc(char *input, wchar_t *wchars)
+{
+	int len = wcstombs(NULL, wchars, 0);
+	if (len >= 0) {
+		input = realloc(input, len + 1);
+		if (input) {
+			if (wcstombs(input, wchars, len) != len) {
+				free(input);
+				input = NULL;
+			} else {
+				input[len] = '\0';
+			}
+		}
+	}
+	return input;
+}
+
+#else /* USE_WIDE_CHAR */
+
+#define wint_t int
+#define wchar_t char
+#define wcstombs(d, s, l) strlen(l)
+#define wchars_to_utf8_str_realloc(i, wc) wc
+#define wget_wch(w, c) ((*(c) = wgetch(w)) ? OK : OK)
+
+#define WACS_VLINE   ACS_VLINE
+#define WACS_DIAMOND ACS_DIAMOND
+#define WACS_UARROW  ACS_UARROW
+#define WACS_DARROW  ACS_DARROW
+#define mvwadd_wchnstr(w, y, x, c, n) mvwaddch(w, y, x, c)
+#define mvwvline_set(w, y, x, c, n) mvwvline(w, y, x, c, n)
+
+#endif /* USE_WIDE_CHAR */
+
+#endif
--- gmu-0.10.1/src/tools/gmuc.c~	2016-07-05 09:14:41.000000000 -0400
+++ gmu-0.10.1/src/tools/gmuc.c	2016-10-23 12:44:47.556039460 -0400
@@ -32,7 +32,7 @@
 #endif
 #include <curses.h>
 #include <signal.h>
-#include <wctype.h>
+#include "charset.h"
 #include <limits.h>
 #include "../wejconfig.h"
 #include "../ringbuffer.h"
@@ -157,23 +157,6 @@
 	return res;
 }
 
-static char *wchars_to_utf8_str_realloc(char *input, wchar_t *wchars)
-{
-	size_t len = wcstombs(NULL, wchars, 0);
-	if (len != (size_t) -1) {
-		input = realloc(input, len + 1);
-		if (input) {
-			if (wcstombs(input, wchars, len) != len) {
-				free(input);
-				input = NULL;
-			} else {
-				input[len] = '\0';
-			}
-		}
-	}
-	return input;
-}
-
 static State do_websocket_handshake(RingBuffer *rb, State state)
 {
 	int  i, data_available = 1;
--- gmu-0.10.1/src/tools/listwidget.c~	2015-12-07 17:36:12.000000000 -0500
+++ gmu-0.10.1/src/tools/listwidget.c	2016-10-23 12:41:19.197189954 -0400
@@ -19,6 +19,7 @@
 #define _XOPEN_SOURCE_EXTENDED
 #endif
 #include <curses.h>
+#include "charset.h"
 #include "window.h"
 #include "listwidget.h"
 
--- gmu-0.10.1/src/tools/ui.h~	2015-11-18 15:19:45.000000000 -0500
+++ gmu-0.10.1/src/tools/ui.h	2016-10-23 12:38:43.102050429 -0400
@@ -17,7 +17,7 @@
 #ifndef WEJ_UI_H
 #define WEJ_UI_H
 
-#include <wchar.h>
+#include "charset.h"
 #include "window.h"
 #include "listwidget.h"
 
